app/helpers/provider/admin/dashboards_helper.rb

```diff
@@ -19,24 +19,21 @@ def sign_class(widget)
     widget.percentual_change > 0 ? 'u-plus' : 'u-minus'
   end
 
-  def dashboard_collection_link(singular_name, collection, path, plural = nil, icon_name = nil)
-    link_to path, class: 'DashboardNavigation-link' do
-      link_text = pluralize(number_to_human(collection.size), singular_name, plural)
+  def dashboard_navigation_link(singular_name, path, collection: nil, icon_name: nil, plural: nil, notice: false)
```

Comment:
The old method was meant for when we had a collection. Shouldn't we add a new method instead of modifying this one? And also extracting what they have in common to a third private method.

Reply:
```ruby
def dashboard_navigation_link(text, path, icon_name: nil, plural: nil, notice: false)
  class_name = "DashboardNavigation-link #{'u-notice' if notice}"
  link_to path, class: class_name do
    link_text = text
    link_text = link_text.prepend "#{icon(icon_name)} " if icon_name
    link_text.html_safe
  end
end

def dashboard_collection_link(singular_name, collection, path, plural = nil, icon_name = nil)
  link_text = pluralize(number_to_human(collection.size), singular_name, plural) 
  dashboard_navigation_link(link_text, path, icon_name, plural, notice)
end
```

Reply:
We would end up with 4 methods and I don't see the win. I like the flexibility of the method that allows for both.

Reply:
I understand that. In the first place I don't think there's much benefit having a method for "collection" links. It's just a wrapper of `pluralize`. Instead, having a "link builder", suitable for all different situations makes the slim code cleaner. And implementation wise is not very complex.